README.md

```diff
@@ -22,7 +63,861 @@ Or install it yourself as:
+
+Policies are separate from preconditions because in we usually treat the results of those checks differently. For example, if an operation fails on preconditions we might want to render a disabled button with an error message as a hint but if the policy fails - we don't render the button at all.
+
+Similarly to preconditions, policies are having 2 returned values signatures: `true/false` and `Success()/Failure(:error_code)`. In case of `false`, an `:unauthorized` error code will be returned as a result. It is possible to return any error code using the `Failure` monad.
+
+```ruby
+class Post::Publish::AuthorPolicy
+  include Dry::Monads[:result]
+
+  def call(post:, current_user:, **)
+    post.author == current_user ? Success() : Failure(:not_an_author)
+  end
+end
+```
+
+It is possible to pass multiple policies, and all of them have to succeed similarly to preconditions. Though it is impossible not to pass any policies for security reasons. If an operation is internal to the system and not exposed to the end user, `policy: nil` should be passed to `Operations::Command` instance anyway just to explicitly specify that this operation doesn't have any policy.
+
+There are 2 more methods to check for policies separately:
+
+* `allowed(**context)` - returns an operation result (success or failure depending on policy checks result). Useful when you need to check the exact error that happened.
+* `allowed?(**context)` - the same as the previous one but returns a boolean result.
+
+### ActiveRecord scopes usage
+
+There might be a temptation to use AR scopes while fetching records from the DB. For example:
+
+```ruby
+class Post::Publish::Contract < Operations::Contract
+  params do
+    optional(:post_id).filled(:integer)
+  end
+
+  rule do |context:|
+    next key.failure(:key?) unless key?(:post_id) && context[:post]
+
+    post = context[:current_user].posts.publishable.find_by(id: values[:post_id])
+
+    if post
+      context[:post] = post
+    else
+      key.failure(:not_found) unless post
+    end
+  end
+end
+```
+
+Please avoid this at all costs since:
+
+1. It is possible to pass context variables like `post` explicitly and there is no guarantee that it will be scoped.
+2. We want to return an explicit `unauthorized/not_publishable` error to the user instead of a cryptic `not_found`.
+3. It does not fit the concept of repositories.
+
+If we want to use security through obscurity in the controller later - we can easily turn a particular operation error into an `ActiveRecord::RecordNotFound` exception at will.
+
+There could be non-domain but rather technical scopes like soft deletion used as an exception but this is a rare case and it should be carefully considered.
+
+### Generic preconditions and policies
+
+Normally we would expect the following execution order of the operation:
+
+1. Set all the context
+2. Check policies
+3. Check preconditions
+4. Validate user input
+5. Call operation if everything is valid
```

Comment:
what about callbacks ?

Reply:
Callbacks are irrelevant in this case and will just confuse.